Firestore/Swift/Source/Result/DocumentReference+Result.swift

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FirebaseFirestore
+
+#if swift(>=5.0)
+
+  extension DocumentReference {
+    /// Writes the document referred to by this `DocumentReference` with the specified data.
+    /// If no document exists, it is created.
+    /// If a document already exists, it is overwritten.
+    ///
+    /// - Parameters:
+    ///   - documentData: The `Dictionary` containing the data for the new document.
+    ///   - completion: The closure to execute on successfully writing to the server or on receipt of an error.
+    ///   It will not be called while the client is offline, though local changes will be visible immediately.
+    ///   - result: The result of request. On success it is empty, otherwise it contains an `Error`.
+    func setData(_ documentData: [String: Any],
+                 completion: @escaping (_ result: Result<Void, Error>) -> Void) {
+      setData(documentData, completion: mapResultClosure(completion))
```

Comment:
It's mildly jarring to read `completion: mapResultClosure(completion)`. Since the API uses `completion` to describe these things, perhaps the function should be `mapResultCompletion` so that this is consistently referred to as a completion handler throughout?

Reply:
Done